src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java

```diff
@@ -124,8 +135,25 @@ public List<CaseEvent> handledEvents() {
     }
 
     private CallbackResponse startClaim(CallbackParams callbackParams) {
+        CaseData.CaseDataBuilder caseDataBuilder = callbackParams.getCaseData().toBuilder();
+        caseDataBuilder.claimStarted(YES);
+
+        if (V_1.equals(callbackParams.getVersion())) {
+            List<LocationRefData> locations = fetchLocationData(callbackParams);
+
+            caseDataBuilder
+                .courtLocation(CourtLocation.builder()
+                   .applicantPreferredCourtLocationList(courtLocationUtils.getLocationsFromList(locations))
+                   .build());
+        }
+
         return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(CaseData.builder().claimStarted(YES).build().toMap(objectMapper)).build();
+            .data(caseDataBuilder.build().toMap(objectMapper)).build();
+    }
+
+    private List<LocationRefData> fetchLocationData(CallbackParams callbackParams) {
```

Comment:
This method is used here and in ResponseCallbackHandler. Can you move it to a service? It is maybe not appropriate to use callbackParams in a service method, but it can receive just the String authToken
That way it is defined once instead of twice in this PR and twice more in mine